tests/components/homeassistant/test_scene.py

```diff
@@ -117,3 +130,82 @@
     assert scene.name == "hallo_2"
     assert scene.state == "scening"
     assert scene.attributes.get("entity_id") == ["light.kitchen"]
+
+
+async def test_snapshot_service(hass, caplog):
+    """Test the snapshot option."""
+    assert await async_setup_component(hass, "scene", {"scene": {}})
+    hass.states.async_set("light.my_light", "on", {"hs_color": (345, 75)})
+    assert hass.states.get("scene.hallo") is None
+
+    assert await hass.services.async_call(
+        "scene",
+        "create",
+        {"scene_id": "hallo", "snapshot_entities": ["light.my_light"]},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+    scene = hass.states.get("scene.hallo")
+    assert scene is not None
+    assert scene.attributes.get("entity_id") == ["light.my_light"]
+
+    hass.states.async_set("light.my_light", "off", {"hs_color": (123, 45)})
+    turn_on_calls = async_mock_service(hass, "light", "turn_on")
+    assert await hass.services.async_call(
+        "scene", "turn_on", {"entity_id": "scene.hallo"}, blocking=True
+    )
+    await hass.async_block_till_done()
+    assert len(turn_on_calls) == 1
+    assert turn_on_calls[0].data.get("entity_id") == "light.my_light"
+    assert turn_on_calls[0].data.get("hs_color") == (345, 75)
+
+    assert await hass.services.async_call(
+        "scene",
+        "create",
+        {"scene_id": "hallo_2", "snapshot_entities": ["light.not_existent"]},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+    assert hass.states.get("scene.hallo_2") is None
+    assert (
+        "Entity light.not_existent does not exist and therefore cannot be snapshot"
+        in caplog.text
+    )
+
+    assert await hass.services.async_call(
+        "scene",
+        "create",
+        {
+            "scene_id": "hallo_3",
+            "entities": {"light.bed_light": {"state": "on", "brightness": 50}},
+            "snapshot_entities": ["light.my_light"],
+        },
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+    scene = hass.states.get("scene.hallo_3")
+    assert scene is not None
+    assert "light.my_light" in scene.attributes.get("entity_id")
+    assert "light.bed_light" in scene.attributes.get("entity_id")
+
+
+async def test_ensure_no_intersection(hass):
+    """Test that entities and snapshot_entities do not overlap."""
+    assert await async_setup_component(hass, "scene", {"scene": {}})
+
+    try:
+        assert await hass.services.async_call(
+            "scene",
+            "create",
+            {
+                "scene_id": "hallo",
+                "entities": {"light.my_light": {"state": "on", "brightness": 50}},
+                "snapshot_entities": ["light.my_light"],
+            },
+            blocking=True,
+        )
+        await hass.async_block_till_done()
+        assert False
+    except vol.MultipleInvalid as ex:
```

Comment:
Use `pytest.raises` context manager instead.

http://doc.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions

Reply:
Thanks. I was looking for something like this :+1:

Reply:
Done. Please excuse the circumstances :see_no_evil: